creator-node/src/services/stateMachineManager/stateReconciliation/updateReplicaSet.jobProcessor.ts

```diff
@@ -592,7 +592,16 @@ const _issueUpdateReplicaSetOp = async (
       throw new Error(errorMsg)
     }
 
-    if (!issueReconfig) {
+    /**
+     * TODO: Remove this after rollout. This is an extra gating condition to NOT issue reconfigs of primaries
+     * except when the cause of the reconfig is that the primary endpoint was deregistered or changed.
+     */
+    const causeOfReconfigIsPrimaryChanged =
```

Comment:
@theoilie is my understanding correct? it's doing 

!!ContentNodeInfoManager.getCNodeEndpointToSpIdMap()[primary] -> does this exist in the cnode to spId mapping? true if exists, false if it doesn't

 const dontIssueEvenIfEnabled = causeOfReconfigIsPrimaryChanged && reconfigType === RECONFIG_MODES.PRIMARY_AND_OR_SECONDARIES -> if cnode endpoint exists and reconfig mode is enabled, set to true
 
if (!issueReconfig || dontIssueEvenIfEnabled) { -> early exit if cnode endpoint exists in mapping and reconfig mode is true


so basically the only condition it would get past the if statement is if the node was deregistered and there was no record in the getCNodeEndpointToSpIdMap

Reply:
yep exactly!